src/components/side_panel/conditional_formatting/conditional_formatting.ts

```diff
@@ -46,18 +46,18 @@ export class ConditionalFormattingPanel extends Component<Props, SpreadsheetChil
       }
     }
     onWillUpdateProps((nextProps: Props) => {
-      const newActiveSheetId = this.env.model.getters.getActiveSheetId();
-      if (newActiveSheetId !== this.activeSheetId) {
-        this.activeSheetId = newActiveSheetId;
-        this.switchToList();
-      } else if (nextProps.selection !== this.props.selection) {
-        const sheetId = this.env.model.getters.getActiveSheetId();
-        const rules = this.env.model.getters.getRulesSelection(sheetId, nextProps.selection || []);
-        if (rules.length === 1) {
-          const cf = this.conditionalFormats.find((c) => c.id === rules[0]);
-          if (cf) {
-            this.editConditionalFormat(cf);
-          }
+      if (this.state.mode == "list") {
+        this.activeSheetId = this.env.model.getters.getActiveSheetId();
```

Comment:
`this.activeSheetId` is useless no ? it's only used when we're in lsit mode, and this make sure that it's always equal to `getters.getActiveSheetId`. Just use `getters.getActiveSheetId` directly.

Reply:
It's also used as the defaultSheetId for transforming stringXC to range on save (not in list mode, so still to the value of the rule "main" sheet.